src/renderer/components/ActivationKeyForm.tsx

```diff
@@ -0,0 +1,111 @@
+import { Box } from "@material-ui/core";
+import { ipcRenderer, shell } from "electron";
+import { GraphQLClient } from "graphql-request";
+import React, { useCallback, useEffect, useState } from "react";
+import { get, NodeInfo } from "src/config";
+import { getSdk } from "src/generated/graphql-request";
+import Button from "./ui/Button";
+import TextField from "./ui/TextField";
+
+interface Props {
+  onSubmit: (data: FormData) => void;
+}
+
+export interface FormData {
+  activationKey: string;
+}
+
+const ACTIVATION_KEY_REGEX = /^[0-9a-f]+\/[0-9a-f]{40}$/;
+
+type ActivationStatus =
```

Comment:
Making this type as enum rather than string literal union would be likely preferred, as we can separate message itself from state value itself, making editing message more reliable.

Reply:
I'm a little bit against using enums in typescript because of this [post](https://dev.to/dvddpl/whats-the-problem-with-typescript-enums-2okj), and the message is not used in this case.